src/handlers/outbound/thirdpartyTransaction/partyLookup.ts

```diff
@@ -0,0 +1,79 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the "License")
+ and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed
+ on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Sridhar Voruganti <sridhar.voruganti@modusbox.com>
+
+ --------------
+ ******/
+import { StateResponseToolkit } from '~/server/plugins/state'
+import {
+  PISPTransactionModelConfig,
+  PISPTransactionData,
+  ThirdpartyTransactionPartyLookupRequest
+} from '~/models/pispTransaction.interface'
+import {
+  PISPTransactionModel,
+  create
+} from '~/models/pispTransaction.model'
+import { Request, ResponseObject } from '@hapi/hapi'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function post (_context: any, request: Request, h: StateResponseToolkit): Promise<ResponseObject> {
+  const payload = request.payload as ThirdpartyTransactionPartyLookupRequest
+
+  // prepare model data
+  const data: PISPTransactionData = {
+    transactionRequestId: payload.transactionRequestId,
+    payeeRequest: payload.payee,
+    currentState: 'start'
+  }
+
+  // prepare model config
+  const modelConfig: PISPTransactionModelConfig = {
+    kvs: h.getKVS(),
+    pubSub: h.getPubSub(),
+    key: payload.transactionRequestId,
+    logger: h.getLogger(),
+    thirdpartyRequests: h.getThirdpartyRequests(),
+    mojaloopRequests: h.getMojaloopRequests()
+  }
+
+  // create model
+  const model: PISPTransactionModel = await create(data, modelConfig)
```

Comment:
What happens when we call `create()` but with an already existing `transactionRequestId`?

Reply:
good catch @lewisdaly -> if we call partyLookup endpoint and we already have an ongoing flow for a given transactionRequestId it will effect in overwriting the state - witch could bring unexpected side-effects -> there is a need to extend all flow-related handlers by adding proper guards to protect the flow and order of execution

Reply:
I extended the story by extra task dedicated to this finding. I will take care of these guards tomorrow.

Reply:
@lewisdaly I think there is only a need to create a guard for `create`. The transitions are guarded by the state machine so everything works ok when we `load`. What status code in return do you suggest in the case when the request to lookupParty for already used transactionRequestId is done: 418 or 422?